### PR TITLE
Issue/62 update columns

### LINKED ISSUE #62: Update validation csv
## Detailed Description
Update columns to https://github.com/openclimatefix/predict_pv_yield/pull/96#issuecomment-981546725

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Update saved validation results

Columns:

t0_datetime_utc
target_datetime_utc :The datetime the forecast is about. i.e. t0_datetime_utc + (forecast_timestep_i * forecast_sample_period)
forecast_gsp_pv_outturn_mw
actual_gsp_pv_outturn_mw
gsp_id
batch_index (optional, but useful for debugging)
example_index (optional, but useful for debugging)

Fixes #62 

## How Has This Been Tested?

add unittests

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_utils/metrics/validation.py [hunk 1]
```diff
@@ -9,7 +10,12 @@
 
 
 def make_validation_results(
-    predictions, truths, gsp_ids, t0_datetimes_utc, batch_idx: Optional[int] = None
+    predictions_mw,
+    truths_mw,
+    gsp_ids: List[int],
+    t0_datetimes_utc: pd.DatetimeIndex,
+    batch_idx: Optional[int] = None,
+    forecast_sample_period: timedelta = timedelta(minutes=30),
```
Comment (JackKelly) [inline nowcasting_utils/metrics/validation.py:18]: Regarding the docstring:

Please update the description of the columns that are made

Also, please add docstring info about the expected shape of `predictions_mw` and `truths_mw`.

File: nowcasting_utils/metrics/validation.py [hunk 2]
```diff
@@ -27,18 +33,30 @@ def make_validation_results(
...
 
-    predictions = pd.DataFrame(
-        predictions, columns=[f"prediction_{i}" for i in range(predictions.shape[1])]
-    )
-    truths = pd.DataFrame(truths, columns=[f"truth_{i}" for i in range(truths.shape[1])])
+    assert predictions_mw.shape == truths_mw.shape
+
+    results_per_forecast_horizon = []
+
+    for i in range(predictions_mw.shape[1]):
```
Comment (JackKelly) [inline nowcasting_utils/metrics/validation.py:40]: To make it really explicit what the loop is looping round, please consider doing something like:

```python
n_forecast_timesteps = preidctions_mw.shape[1]
for forecast_timestep_i in range(n_forecast_timesteps):
```

(although I'm not sure if `predictions_mw.shape` is the number of timesteps or something else!  So please adapt as appropriate!)

Also:  If this loop is slow, then could this be done in a "vectorised" way (although I'm not entirely sure how!  Maybe using `pd.DataFrame.pivot`?)

**Review by JackKelly — APPROVED** [review]
LGTM! Thanks!

File: nowcasting_utils/metrics/validation.py (hunk 2 above)
Comment (peterdudfield) [reply nowcasting_utils/metrics/validation.py:40]: I did want to do it in a vectorizd way, but I wasnt sure sure how. I'll perhaps leave that one for another time. 

But yea making the loop more clear is a good shout

**Pull request merged**